pkg/service/controller_test.go

```diff
@@ -50,33 +50,46 @@ func TestCreateVolume_CustomBus(t *testing.T) {
 	assert.Equal(t, response.GetVolume().GetVolumeContext()[busParameter], string(*testBusType))
 }
 
+func TestCreateVolume_NotAllowedStorageClass(t *testing.T) {
+	client := &ControllerClientMock{t: t}
+	allowedStorageClasses = []string{"allowedClass"}
+	allowAllStorageClasses = false
+	controller := ControllerService{client, testInfraNamespace, testInfraLabels, allowedStorageClasses, allowAllStorageClasses}
+
+	request := getCreateVolumeRequest()
+	request.Parameters[InfraStorageClassNameParameter] = "notAllowedClass"
+
+	_, err := controller.CreateVolume(context.TODO(), request)
+	assert.NotNil(t, err)
```

Comment:
can we check that the expected err msg is received here? That simply confirms that it was your validation check that actually prevented the request from succeeding.